internal/saucecloud/cloud.go

```diff
@@ -336,11 +340,28 @@ func (r *CloudRunner) runJobs(jobOpts chan job.StartOptions, results chan<- resu
 			continue
 		}
 
+		if opts.CurrentPassCount < opts.PassThreshold && opts.Attempt < opts.Retries {
+			opts.Attempt++
+			jobOpts <- opts
```

Comment:
Would cause a deadlock. Needs same fix as https://github.com/saucelabs/saucectl/pull/650.

Having said that, can't this block be combined with the re-schedule above somehow? It's nigh identical in code, except for one log line.

Reply:
done